### DISCUSSION TITLE
Road bike route go through dirt road (highway=path)

### DISCUSSION THREAD

**Discussion by nemoneph** [valhalla/valhalla#3799] [Q&A]
Hello, 
I have a problem with a road bike route:

  'costing' => 'bicycle'
  'bicycle_type' => 'Road'
  
The route currently goes on path through dirt road (highway=path)
It's OK for mountain bike but not for road/racing bike, so I would like to exclude this tag on my search.

I don't see a way to do that on the API reference.
Someone has an idea for this problem ?

PS : on graphhopper, it's seem OK, it does not go through the dirt road when choose a "racingbike".

Thanks a lot :)

**Comment by kevinkreiser** [comment]
@nemoneph you'll need to provide us with the road in question so we can check why it wouldnt avoid it. The road bike is supposed to avoid those types of roads but without any clue which road it is its impossible for us to know why. please share a bit more about where this is happening.

**Comment by dnesbitt61** [comment]
Try adding the bicycle costing option avoid_bad_surfaces (see API docs) and set it to 1. That should avoid all paths when using a road bike. The default is to penalize them but allow them so that short sections of unpaved roads are allowed when road bicycle type is selected.

**Reply by nemoneph** [reply]
@dnesbitt61 Thx, but already tried with this option.
 Didn't change anything.

**Comment by nemoneph** [comment]
@kevinkreiser Yes I understand.

Here an example for this trace :

https://valhalla.openstreetmap.de/directions?profile=bicycle&wps=5.134778022766114%2C44.89476536329048%2C5.158038139343263%2C44.88692126319926

![screen](https://user-images.githubusercontent.com/3691890/196941915-c713d441-af54-4c00-b366-7d9e3aad9a6d.png)

Payload : 

`{"costing":"bicycle","costing_options":{"bicycle":{"exclude_polygons":[],"maneuver_penalty":5,"country_crossing_penalty":0,"country_crossing_cost":600,"use_ferry":1,"use_living_streets":0.5,"service_penalty":15,"service_factor":1,"shortest":false,"bicycle_type":"Road","cycling_speed":20,"use_roads":0.5,"use_hills":0.5,"avoid_bad_surfaces":1}},"exclude_polygons":[],"locations":[{"lon":5.134778022766114,"lat":44.89476536329048},{"lon":5.158038139343263,"lat":44.88692126319926}],"directions_options":{"units":"kilometers"},"id":"valhalla_directions"}`


Here the OSM portion with dirty road "path" : 

https://www.openstreetmap.org/way/345274274#map=19/44.89248/5.13878
OR
https://www.openstreetmap.org/way/345274274

**Reply by kevinkreiser** [reply]
its only tagged as `path` which doesnt tell us about the surface exactly.

from the osm wiki, https://wiki.openstreetmap.org/wiki/Tag:highway%3Dpath:
> The path may have any type of surface

if you want us to avoid that path, you can tag it with the correct surface tag and the road bike will avoid it.

**Reply by nemoneph** [reply]
Okay thx for the info.

Indeed I can add and I will add the surface tag for this example, it's good for improve OSM.
But this is one example among many others with this case (path without surface).

**Comment by dnesbitt61** [comment]
With the data I am using, that way (highway=path) has surface = kCompacted. Without other OSM surface tags this is the surface type that is assigned - and it is not penalized it seems due to this:

```c++
  // If surface is worse than the minimum we add a surface factor
  if (edge->surface() >= minimal_surface_penalized_) {
    factor +=
        avoid_bad_surfaces_ * kSurfaceFactors[static_cast<uint32_t>(edge->surface()) -
                                              static_cast<uint32_t>(minimal_surface_penalized_)];
  }
```

This code is not 100% correct. For road bikes it is setting kCompacted to the minimal_surface_penalized_, but the subtraction ends up being 0 for kCompacted edge->surface().

**Reply by nemoneph** [reply]
ah .. it's explain why avoid_bad_surfaces didn't have any effect for this.

**Comment by johnburgg** [comment]
Hello,

I understand your concern about the road bike route including a dirt road section. To address this, you might want to consider using specific tags or filters to customize your route preferences. While the API reference might not explicitly mention excluding certain road types, you can often achieve this by refining your search parameters.

To get more detailed assistance and explore potential solutions, I recommend visiting a helpful resource like [Tribi cycle](https://tribicycle.com) They provide valuable information about bike routes and can guide you on how to adjust your route preferences to ensure a smooth road biking experience. Exploring their guides and forums might provide insights from experienced cyclists who have faced similar challenges. Good luck with your road bike route, and I hope you find the solution that suits your needs!
